src/hooks/api/getters/useFeatures/useFeatures.ts

```diff
@@ -1,40 +1,39 @@
 import useSWR, { mutate, SWRConfiguration } from 'swr';
-import { useState, useEffect } from 'react';
+import { useCallback } from 'react';
 import { formatApiPath } from '../../../../utils/format-path';
 import handleErrorResponses from '../httpErrorResponseHandler';
+import { IFeatureToggle } from '../../../../interfaces/featureToggle';
 
-const useFeatures = (options: SWRConfiguration = {}) => {
-    const fetcher = async () => {
-        const path = formatApiPath('api/admin/features/');
-        return fetch(path, {
-            method: 'GET',
-        })
-            .then(handleErrorResponses('Features'))
-            .then(res => res.json());
-    };
-
-    const FEATURES_CACHE_KEY = 'api/admin/features/';
-
-    const { data, error } = useSWR(FEATURES_CACHE_KEY, fetcher, {
-        ...options,
-    });
+const PATH = formatApiPath('api/admin/features');
 
-    const [loading, setLoading] = useState(!error && !data);
+export interface IUseFeaturesOutput {
+    features: IFeatureToggle[];
+    refetchFeatures: () => void;
+    loading: boolean;
+    error?: Error;
+}
 
-    const refetchFeatures = () => {
-        mutate(FEATURES_CACHE_KEY);
-    };
+export const useFeatures = (options?: SWRConfiguration): IUseFeaturesOutput => {
+    const { data, error } = useSWR<{ features: IFeatureToggle[] }>(
+        PATH,
+        fetchFeatures,
+        options
+    );
 
-    useEffect(() => {
-        setLoading(!error && !data);
-    }, [data, error]);
+    const refetchFeatures = useCallback(() => {
+        mutate(PATH).catch(console.warn);
+    }, []);
 
     return {
         features: data?.features || [],
-        error,
-        loading,
+        loading: !error && !data,
         refetchFeatures,
+        error,
     };
 };
 
-export default useFeatures;
+function fetchFeatures() {
```

Comment:
Could be arrow for consistency :)

But I like this pattern, we should move the fetcher and path outside of the hook in every instance where it's not dependent on input parameters.